### PR TITLE
Jkmarx/add primary group api

### PR DESCRIPTION
**Pull request by jkmarx**
-API to support user-profile updates, specifically the primary-group
Ref #2781

### CODE REVIEW

File: refinery/core/serializers.py [hunk 1]
```diff
@@ -62,6 +62,38 @@ def partial_update(self, instance, validated_data):
...
+    def validate_primary_group(self, group):
+        user = self.context.get('request').user
+        if user.id in group.user_set.values_list('id', flat=True):
+            pass
+        else:
+            raise serializers.ValidationError(
+                'User is not a member of group, %s', group
+            )
+
+        if group.name != 'Public':
```
Comment (scottx611x) [inline refinery/core/serializers.py]: `settings.REFINERY_PUBLIC_GROUP_NAME`

File: refinery/core/views.py [hunk 2]
```diff
@@ -1079,3 +1080,49 @@ def post(self, request):
         token["Region"] = region
 
         return Response(token)
+
+
+class UserProfileViewSet(APIView):
+    """API endpoint that allows for UserProfiles to be edits.
```
Comment (scottx611x) [inline refinery/core/views.py]: `to be edited`

File: refinery/core/views.py [hunk 3]
```diff
@@ -1079,3 +1080,49 @@ def post(self, request):
...
+              description: group id
+              type: int
+              paramType: form
+              required: false
+    ...
+    """
+    http_method_names = ["patch"]
+
+    def patch(self, request, uuid):
+        if not request.user.is_anonymous():
```
Comment (scottx611x) [inline refinery/core/views.py]: What about flipping the first ifs logic here:
```
def patch(self, request, uuid):
    if request.user.is_anonymous():
        return Response(
            self.user, status=status.HTTP_401_UNAUTHORIZED
        )
    serializer = UserProfileSerializer(request.user.profile,
                                       data=request.data,
                                       partial=True,
                                       context={'request': request})
    if serializer.is_valid():
        serializer.save()
        return Response(
            serializer.data, status=status.HTTP_202_ACCEPTED
        )
    return Response(
        serializer.errors, status=status.HTTP_400_BAD_REQUEST
    )
```

File: refinery/core/migrations/0020_create_initial_site_statistics.py [hunk 4]
```diff
@@ -22,7 +21,9 @@ def create_initial_site_statistics(apps, schema_editor):
             [dataset for dataset in DataSet.objects.all() if dataset.shared]
```
[the remark this answers is not in the document]
Comment (jkmarx) [reply refinery/core/migrations/0020_create_initial_site_statistics.py]: sorry, I forgot to push my update.

**Pull request merged**